gpii/node_modules/eventLog/src/eventLog.js

```diff
@@ -228,73 +257,86 @@ gpii.eventLog.gotError = function (err, errType) {
-        logPath = process.env.GPII_EVENT_LOG || that.logFilePath || that.options.logDestination;
-    }
+gpii.eventLog.initLogDestination = function (that, logPath) {
+    logPath = process.env.GPII_EVENT_LOG || logPath;
 
-    if (logPath) {
-        if (logPath.startsWith("tcp:")) {
-            var m = /tcp:\/*([^:]+):([0-9]+)/.exec(logPath);
-            if (m) {
-                that.logHost = m[1];
-                that.logPort = m[2];
-                that.logFilePath = null;
-            }
-        } else {
-            that.logHost = null;
-            that.logFilePath = logPath;
-        }
+    var match = logPath && /tcp:\/*([^:]+):([0-9]+)/.exec(logPath);
+    if (match) {
+        that.logServer.host = match[1];
+        that.logServer.port = match[2];
     } else {
-        that.logHost = null;
-        var startupTime = Date.now();
-        var gpiiSettingsDir = process.env.GPII_EVENT_LOG_DIRECTORY || that.getGpiiSettingsDir();
-        that.logFilePath = path.join(gpiiSettingsDir, "gpii-" + gpii.journal.formatTimestamp(startupTime) + ".log");
+        that.logPath = logPath;
     }
 
-    var dest = that.logHost ? ("tcp://" + that.logHost + ":" + that.logPort) : that.logFilePath;
+    var dest = that.logServer.host ? ("tcp://" + that.logServer.host + ":" + that.logServer.port) : that.logPath;
     fluid.log(fluid.logLevel.IMPORTANT, "Writing event log to " + dest);
-
-    return that.logFilePath;
 };
 
 /**
  * Writes an event to the log file.
  *
  * @param {Component} that - The gpii.eventLog instance.
  * @param {Object} level - Level of the log, see fluid.logLevelsSpec [FATAL,FAIL,WARN,IMPORTANT,INFO,TRACE].
- * @param {Object} event - The object. This will be modified to what has been sent to the log, adding the installID and
+ * @param {LogEvent} event - The object. This will be modified to what has been sent to the log, adding the installID and
  * timestamp fields.
  */
 gpii.eventLog.writeLog = function (that, level, event) {
-    var intLevel = gpii.eventLog.checkLevel(level);
-    event.level = intLevel.value;
+    var eventLevel = gpii.eventLog.checkLevel(level);
+    event.level = eventLevel.value;
 
-    // Log to console before the installation ID and timestamp are added (no one wants to see it).
-    fluid.log(fluid.logLevel.TRACE, event);
+    gpii.eventLog.recordDuration(that, event);
 
-    event.installID = that.installationID;
+    Object.assign(event, that.eventData);
     event.timestamp = gpii.eventLog.getTimestamp();
-    event.version = that.version;
     event.sequence = that.sequence++;
 
-    var logLine = JSON.stringify(event) + "\n";
+    if (eventLevel.priority <= that.logLevel.priority) {
+        var logLine = JSON.stringify(event) + "\n";
+
+        if (that.logServer.host && that.logServer.port) {
+            gpii.eventLog.writeLogTcp(that, logLine);
+        }
 
-    if (that.logHost && that.logPort) {
-        gpii.eventLog.writeLogTcp(that, logLine);
+        if (that.logPath) {
```

Comment:
Since the destination can either be a file or tcp, wouldn't using an else block (with a proper comment) be more accurate here?

Reply:
fixed